crates/repl/src/kernels.rs

```diff
@@ -465,6 +467,123 @@ async fn read_kernels_dir(path: PathBuf, fs: &dyn Fs) -> Result<Vec<LocalKernelS
     Ok(valid_kernelspecs)
 }
 
+#[allow(dead_code)]
+pub fn python_env_kernel_specifications(
+    project: &Model<Project>,
+    worktree_id: WorktreeId,
+    cx: &mut AppContext,
+) -> impl Future<Output = Result<Vec<KernelSpecification>>> {
+    let fs = project.read(cx).fs().clone();
+    let python_language = LanguageName::new("Python");
+
+    let toolchains = project
+        .read(cx)
+        .available_toolchains(worktree_id, python_language, cx);
+
+    async move {
+        let toolchains = toolchains.await;
+
+        let toolchains = if let Some(toolchains) = toolchains {
+            toolchains
+        } else {
+            return Ok(Vec::new());
+        };
+
+        let mut kernel_specs = Vec::new();
+
+        for toolchain in toolchains.toolchains {
+            let python_path = toolchain.path.to_string();
+
+            // Check if ipykernel is installed
+            let ipykernel_check = Command::new(&python_path)
+                .args(&["-c", "import ipykernel"])
+                .output()
+                .await;
+
+            if ipykernel_check.is_ok() && ipykernel_check.unwrap().status.success() {
+                // Create a default kernelspec for this environment
+                let default_kernelspec = JupyterKernelspec {
+                    argv: vec![
+                        python_path.clone(),
+                        "-m".to_string(),
+                        "ipykernel_launcher".to_string(),
+                        "-f".to_string(),
+                        "{connection_file}".to_string(),
+                    ],
+                    display_name: toolchain.name.to_string(),
+                    language: "python".to_string(),
+                    interrupt_mode: None,
+                    metadata: None,
+                    env: None,
+                };
+
+                kernel_specs.push(KernelSpecification::PythonEnv(LocalKernelSpecification {
+                    name: toolchain.name.to_string(),
+                    path: PathBuf::from(&python_path),
+                    kernelspec: default_kernelspec,
+                }));
+            } else {
+                continue;
+            }
+
+            let prefix = PathBuf::from(toolchain.path.to_string());
```

Comment:
Delete this code now that we're just collecting pure python kernelspecs. We can bring back the kernelspecs-in-env code (which would likely support R kernels in conda) once we have the `prefix` from the python toolchains.